util/topgen/c.py

```diff
@@ -52,13 +52,20 @@ def __init__(self, name, base_addr, size_bytes):
         self.name = name
         self.base_addr = base_addr
         self.size_bytes = size_bytes
+        self.size_words = int(str(size_bytes), 0) // 4
```

Comment:
what type is `size_bytes` here? A string? An int? What are these casts achieving?

Reply:
I will unresolve the [comment](https://github.com/lowRISC/opentitan/pull/3576#discussion_r500337861). The short answer is, I guess something that uses it passes that parameter as a string (don't remember exactly where, but if you run `make -C hw top` then without this cast magic - this line will throw an error). Potentially the fix is simple, but I haven't looked at the cause of the problem.

Reply:
I suggest casting both `base_addr` and `size_bytes` to an integer  `TopGenC.modules()` and `TopGenC.memories()`. That's the moment that we pluck a string out of the dictionary of hjson data, so is probably the right time to "parse" things.

Reply:
In that case that would need `hex(int(var, 0))` to keep the hex representation in `top_earlgrey.h`.

Reply:
Yeah, I ran it and saw that too. But that's fine: the thing looks like a number in the configuration file; I think it's fine for the scripts to decide how to format it.